test/test_SW_Flow_Lib.cc

```diff
@@ -308,7 +308,1074 @@ TEST(SWFlowTest, LitterInterceptedWater) {
+    Reset_SOILWAT2_after_UnitTest();
+
+    //Begin TEST for if swc[j] > swcsat[j]
+    //INPUTS
+    double swc2_1[1] = {1.02};
+    double swcsat2_1[1] = {1.01};
+
+    //INPUTS for expected outputs
+    double swcExpected4_1[1] = {0};
+    double drainExpected4_1[1] = {1.589};
+    standingWaterExpected = 0.011;
+    drainoutExpected = 0.689;
+
+    infiltrate_water_low(swc2_1, drain_1, &drainout, nlyrs, sdrainpar, sdraindpth, swcfc_1, width_1, swcmin_1,
+      swcsat2_1, impermeability_1, &standingWater);
+
+    EXPECT_NEAR(drainout, drainoutExpected, 0.0000001); //drainout is expected to be 0.67
+    EXPECT_NEAR(standingWater, standingWaterExpected, 0.0000001); //No standingWater expected
+
+    for(unsigned int i = 0; i < nlyrs; i++){
+
+      EXPECT_NEAR(swc_1[i], swcExpected4_1[i], 0.0000001); //swc is expected to match swcExpected
+      EXPECT_NEAR(drain_1[i], drainExpected4_1[i], 0.0000001); //drain is expected to match drainExpected
+    }
+
+    //Reset to previous global states.
+    Reset_SOILWAT2_after_UnitTest();
+
+    //Begin TEST for if swc[j] <= swcsat[j]
+    //INPUTS
+    double swcsat3_1[1] = {5.0};
+
+    //INPUTS for expected outputs
+    double swcExpected5_1[1] = {0};
+    double drainExpected5_1[1] = {1.589};
+    standingWaterExpected = 0.011;
+
+    infiltrate_water_low(swc_1, drain_1, &drainout, nlyrs, sdrainpar, sdraindpth, swcfc_1, width_1, swcmin_1,
+      swcsat3_1, impermeability_1, &standingWater);
+
+    EXPECT_NEAR(drainout, drainoutExpected, 0.0000001); //drainout is expected to be 0.67
+    EXPECT_NEAR(standingWater, standingWaterExpected, 0.0000001); //No standingWater expected
+
+    for(unsigned int i = 0; i < nlyrs; i++){
+
+      EXPECT_NEAR(swc_1[i], swcExpected5_1[i], 0.0000001); //swc is expected to match swcExpected
+      EXPECT_NEAR(drain_1[i], drainExpected5_1[i], 0.0000001); //drain is expected to match drainExpected
+    }
+
+    //Reset to previous global states.
+    Reset_SOILWAT2_after_UnitTest();
+  }
+
+  //TEST for hydraulic_redistribution when nlyrs = MAX_LAYERS and nlyrs = 1
+  TEST(SWFlowTest, hydraulic_redistribution){
+
+    //INPUTS
+    double swc[25] = {5.01,10.005,10.02,10.03,20.01, 20.02, 20.03, 20.04, 20.01, 20.05, 20.06, 20.07,
+        20.08, 20.09, 20.10, 20.11, 20.12, 20.13, 20.14, 20.15, 20.16, 20.17, 20.18, 20.19, 20.2};
+    double swcwp[25] = {5,10,10,10,20, 20, 20, 20, 20, 20, 20, 20, 20, 20, 20, 20, 20, 20, 20, 20, 20, 20, 20, 20, 20};
+    double lyrRootCo[25] = {.10,.11,.12,.13,.14,.15,.16,.17, .18,.19,.20,.21,.22,.23,.24,.25, .26,.27,.28,.29,.30,.31,.32,.33, .34};
+    double hydred[25] = {1,2,3,4,5,6,7,8, 1,2,3,4,5,6,7,8, 1,2,3,4,5,6,7,8, 9}; //hydred[0] != 0 for test, otherwise these don't matter as they are reset to zero in function
+    unsigned int nlyrs = MAX_LAYERS;
+    double maxCondroot = -0.2328;
+    double swp50 = 1.2e12;
+    double shapeCond = 1;
+    double scale = 0.3;
+
+    // Setup soil layers
+    create_test_soillayers(nlyrs);
```

Comment:
defining these arrays independent from the call to `create_test_soillayers` asks for trouble and unexpected failures if anything changes somewhere else

--> use the approach implemented in `TEST(SWFlowTest, pot_soil_evap)` instead, i.e., extract necessary array values from the soil after the call to `create_test_soillayers`